serialize interval data as JSON

ironcladlou opened the pull request.

paulfantom committed:
test/extended/prometheus: test if AlertmanagerReceiversNotConfigured is firing

deads2k committed:
UPSTREAM: <carry>: simplify the authorizer patch to allow the flags to function

waveywaves committed:
Use --kubeconfig flag for Jenkins Tests

- Changes similar to https://github.com/openshift/origin/commit/4a31327f4aff3842812aae731adba9257e28ee54 as `--config` is deprecated and the replacement for the same is `--kubeconfig`
- https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_jenkins/1015/pull-ci-openshift-jenkins-master-e2e-aws-jenkins/756 is a failed build where we are getting the error.
```
https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_jenkins/1015/pull-ci-openshift-jenkins-master-e2e-aws-jenkins/756
```
- Updating this would allow for the Jenkins tests to be done by using the --kubeconfig flag with oc instead of --config

smarterclayton committed:
test: Test internal port connectivity internally for 9000-9999

Test that both ends of the range are available. Verify point to
point connectivity across the whole cluster.

jsafrane committed:
Enable CSI snapshot tests

soltysh committed:
Bug 1789266: replace deprecated commands with current secrets commands

huffmanca committed:
UPSTREAM: 82027: Enabled reading config files for vsphere e2e tests

ironcladlou committed:
serialize interval data as JSON

paulfantom committed:
test/extended/util/annotate/generated: regenerate

openshift-merge-robot committed:
Merge pull request #24448 from waveywaves/patch-1

Bug 1796483: Use --kubeconfig flag for Jenkins Tests

ironcladlou commented:
https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/24481/pull-ci-openshift-origin-master-e2e-gcp-upgrade/2170/artifacts/e2e-gcp-upgrade/junit/

The contents are correct but the filename is wrong and I'm not sure why.

bertinatto committed:
Bug 1796895: Don't use annotation in openshift/csi test suite

In df6f5bd we added annotated tests to generated bindata. However,
CSI tests are not included there, since they can be installed
on-demand. This commit calls the CSI directly, without using the
annotations.

smarterclayton commented on pkg/test/ginkgo/cmd_runsuite.go:
```diff
@@ -236,6 +238,23 @@ func (opt *Options) Run(args []string) error {
 	var syntheticTestResults []*JUnitTestCase
 	if events := m.Events(time.Time{}, time.Time{}); len(events) > 0 {
+		// Serialize the interval data for easier external analysis
+		intervalsFilename := filepath.Join(opt.JUnitDir, "intervals.json.gz")
```
JUnit dir can be unset.

ironcladlou replied:
Handled

smarterclayton commented on pkg/test/ginkgo/cmd_runsuite.go:
```diff
@@ -236,6 +238,23 @@ func (opt *Options) Run(args []string) error {
+		if file, err := os.Create(intervalsFilename); err != nil {
+			fmt.Fprintf(opt.Out, "ERROR: failed to create %s: %v\n", intervalsFilename, err)
+		} else {
+			gz := gzip.NewWriter(file)
```
Don't gzip automatically, you should just output the raw file.

ironcladlou replied:
I think it's like 3mb vs. 300kb. I removed the compression

smarterclayton commented on pkg/test/ginkgo/cmd_runsuite.go:
```diff
@@ -236,6 +238,23 @@ func (opt *Options) Run(args []string) error {
+		// Serialize the interval data for easier external analysis
+		intervalsFilename := filepath.Join(opt.JUnitDir, "intervals.json.gz")
+		if file, err := os.Create(intervalsFilename); err != nil {
+			fmt.Fprintf(opt.Out, "ERROR: failed to create %s: %v\n", intervalsFilename, err)
```
Error messages go to `opt.ErrOut`, and your prefix should be `error: Failed to create event output: %v` (don't display the filename, display what you're trying to do because the os error usually has the filename).

ironcladlou replied:
thanks, fixed

smarterclayton commented on pkg/test/ginkgo/cmd_runsuite.go:
```diff
@@ -236,6 +238,23 @@ func (opt *Options) Run(args []string) error {
+			if err := json.NewEncoder(gz).Encode(events); err != nil {
+				fmt.Fprintf(opt.Out, "ERROR: failed to write %s: %v\n", intervalsFilename, err)
+			} else {
+				fmt.Fprintf(opt.Out, "wrote monitor interval data to %s\n", intervalsFilename)
```
Don't write this, success doesn't need a message.

smarterclayton commented on pkg/test/ginkgo/cmd_runsuite.go:
```diff
@@ -236,6 +238,23 @@ func (opt *Options) Run(args []string) error {
+			fmt.Fprintf(opt.Out, "ERROR: failed to create %s: %v\n", intervalsFilename, err)
+		} else {
+			gz := gzip.NewWriter(file)
+			defer func() {
```
This defer is going to be super late, you shouldn't use defer here but just have two separate close paths (one for error paths, one for success).

Note you have to close the file anyway to ensure it's flushed.

ironcladlou replied:
Good catch, fixed (using a single close now because I switched to os.OpenFile)

ironcladlou committed:
refactor interval dump

ironcladlou commented:
All feedback addressed, thanks

ironcladlou commented:
Here's how it looks

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24485/pull-ci-openshift-origin-master-e2e-gcp-upgrade/2184

openshift-merge-robot committed:
Merge pull request #24435 from deads2k/simplify-authorizer-patch

bug 1797740: UPSTREAM: <carry>: simplify the authorizer patch to allow the flags t…

openshift-merge-robot committed:
Merge pull request #24455 from smarterclayton/health_check_internal

Test internal port connectivity internally for 9000-9999

p0lyn0mial committed:
UPSTREAM 87778: makes unavailableGauge metric to always reflect the current state of a service

sttts committed:
UPSTREAM: 87813: kube-aggregator: increase log level of AggregationController API group logging

ingvagabund committed:
UPSTREAM: 87242: SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run - set resource requests as well

openshift-merge-robot committed:
Merge pull request #24497 from sttts/sttts-schema-aggregation-logging

Bug 1747479: UPSTREAM: 87813: kube-aggregator: increase log level of AggregationController API group logging

openshift-merge-robot committed:
Merge pull request #24449 from jsafrane/enable-snapshot-tests

Bug 1798131: Enable CSI snapshot tests

smarterclayton committed:
test: Move the prometheus alerts test to the beginning of the suite

Otherwise some tests trigger temporary alerts

openshift-merge-robot committed:
Merge pull request #24498 from ingvagabund/backport-87242

bug 1779303: UPSTREAM: 87242: SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run - set resource requests as well

openshift-merge-robot committed:
Merge pull request #24471 from smarterclayton/service_upgrade

Bug 1796715: Add a service upgrade test that verifies availability

openshift-merge-robot committed:
Merge pull request #24484 from bertinatto/fix-openshift-tests-csi-drivers

Bug 1796895: Don't use annotation in openshift/csi test suite

stlaz committed:
UPSTREAM: <carry>: bootstrap user - make tokens have implicit audiences

openshift-merge-robot committed:
Merge pull request #24277 from paulfantom/test_AlertmanagerReceiversNotConfigured

Bug 1797634: test/extended/prometheus: test if AlertmanagerReceiversNotConfigured is firing

Commit:
Test if oc tag can work without imagestreamtags

openshift-merge-robot committed:
Merge pull request #24503 from stlaz/bootstrapuser_audiences

Bug 1794885: UPSTREAM: <carry>: bootstrap user - make tokens have implicit audiences

smarterclayton committed:
test: Disable Azure OLM etcd operator test due to flakiness

Opened https://bugzilla.redhat.com/show_bug.cgi?id=1799491 to track
identifying and fixing why.

rphillips committed:
UPSTREAM: 84614: kubelet: add certificate rotation error metric

openshift-merge-robot committed:
Merge pull request #24466 from soltysh/remove_deprecated_secrets

Bug 1789266: replace deprecated commands with current secrets commands

openshift-merge-robot committed:
Merge pull request #24496 from p0lyn0mial/pick-unavailableGauge-87778

Bug 1798450: makes unavailableGauge metric to always reflect the current state of a service

openshift-merge-robot committed:
Merge pull request #24513 from rphillips/backport/84614

Bug 1767523: UPSTREAM: 84614: kubelet: add certificate rotation error metric

openshift-merge-robot committed:
Merge pull request #24512 from smarterclayton/skip_azure

Bug 1799491: Disable Azure OLM etcd operator test due to flakiness

openshift-merge-robot committed:
Merge pull request #24499 from smarterclayton/smoke_alerts

test: Move the prometheus alerts test to the beginning of the suite

openshift-merge-robot committed:
Merge pull request #24478 from huffmanca/read_vsphere_config

Bug 1722672: UPSTREAM: 82027: Enabled reading config files for vsphere e2e tests

openshift-merge-robot committed:
Merge pull request #24505 from dmage/oc-tag-tests

Bug 1746149: test if oc tag can work without imagestreamtags
